coremltools/models/utils.py

```diff
@@ -1675,3 +1681,394 @@ def randomize_weights(mlmodel: "_ct.models.MLModel"):
     )
 
     return randomized_mlmodel
+
+
+def bisect_model(
+    model_path: str,
+    output_dir: str,
+    remove_original: _Optional[bool] = False,
+    merge_chunks_to_pipeline: _Optional[bool] = False,
+    check_output_correctness: _Optional[bool] = False,
+):
+    """
+    Utility function to split a mlpackage model into two mlpackages of approximately same file size.
+
+    The implementation of this util is mainly migrated from the below source:
+    https://github.com/apple/ml-stable-diffusion/blob/main/python_coreml_stable_diffusion/chunk_mlprogram.py
+
+    Parameters
+    ----------
+    model_path: str
```

Comment:
Should we also accept an mlmodel instance here? I'd say allowing both is pretty standard in an our API.

Reply:
+1

Reply:
Done!